File: backend/routers/prediction.py
from fastapi import APIRouter, Depends
from routers.auth import get_current_user
from services import prediction_service
from models.schemas import PredictionRequest

router = APIRouter(prefix="/prediction", tags=["prediction"])


@router.get("/districts")
def districts(_=Depends(get_current_user)):
    return prediction_service.get_districts()


@router.post("/predict")
def predict(payload: PredictionRequest, _=Depends(get_current_user)):
    return prediction_service.predict(
        payload.district,
        payload.rainfall,
        payload.temperature,
        payload.month,
        payload.day,
    )

File: frontend/src/api/prediction.ts
import axios from "axios";
import { useAuthStore } from "../store/authStore";

const BASE = "https://twinnetra-app.onrender.com";
const authHeader = () => ({
  Authorization: `Bearer ${useAuthStore.getState().token}`,
});

export const fetchDistricts = () =>
  axios.get(`${BASE}/prediction/districts`, { headers: authHeader() }).then((r) => r.data);

export const runPrediction = (payload: {
  district: string;
  rainfall: number;
  temperature: number;
  month: number;
  day: number;
}) =>
  axios
    .post(`${BASE}/prediction/predict`, payload, { headers: authHeader() })
    .then((r) => r.data);
